src/groupeddataframe/grouping.jl

```diff
@@ -1064,10 +1146,16 @@ function _combine(f::Any, gd::GroupedDataFrame)
     end
     agg = check_aggregate(fun)
     if agg isa AbstractAggregate && f isa Pair{<:ColumnIndex}
-        idx = gd.idx[gd.starts]
+        idx = Vector{Int}(undef, length(gd))
+        # Find a representative row for each group
+        # TODO: see whether short-circuiting would be possible without slowing down the loop
+        @inbounds for (i, g_ix) in enumerate(gd.groups)
+            g_ix > 0 && (idx[g_ix] = i)
```

Comment:
maybe you can write `idx[g_ix] = ifelse(g_ix > 0, i , 0)`. This should allow avoiding branching in this loop and enable SIMD (I am not sure how performance critical this part of code is).

Reply:
Unfortunately no, since the index would be `0` in that case. One approach I have tried is doing `g_ix+1`, allocating a vector with one more entry, and resizing it afterwards. But anyway there are no SIMD instructions for this operation on most CPUs (I think some AVX512 machines have them, but I haven't been able to find one to test), so it doesn't make a difference for performance.

Reply:
OK